storage/snapshot.go

```diff
@@ -6,6 +6,9 @@ package storage
 type Snapshot struct {
 	Name    string // The snapshot name or other identifier you would use to reference it
 	Created string // The UTC time that the snapshot was created, in RFC3339 format
+	ID      string // Unique ID assigned to the snapshot at the time of creation
+	Backend string // Name of the storage backend
+	Pool    string // Name of the pool on which the source volume and this snapshot were provisioned
```

Comment:
A snapshot struct should not need a Pool attribute.  Pools are only used to match storage classes with available storage resources while provisioning volumes.  Once created, a volume may always be located using only the backend & volume names.  Therefore, a volume snapshot should always be addressable by backend, volume, & snapshot name.